src/main/java/com/lucidworks/spark/SolrRelation.java

```diff
@@ -223,23 +265,566 @@ protected String fq(Filter f) {
   }
 
   public void insert(final DataFrame df, boolean overwrite) {
-    JavaRDD<SolrInputDocument> docs = df.javaRDD().map(new Function<Row, SolrInputDocument>() {
-      public SolrInputDocument call(Row row) throws Exception {
-        StructType schema = row.schema();
-        SolrInputDocument doc = new SolrInputDocument();
-        for (StructField f : schema.fields()) {
-          String fname = f.name();
-          if (fname.equals("_version_"))
-            continue;
-
-          int fieldIndex = row.fieldIndex(fname);
-          Object val = row.isNullAt(fieldIndex) ? null : row.get(fieldIndex);
-          if (val != null)
-            doc.setField(fname, val);
-        }
-        return doc;
+    JavaRDD<SolrInputDocument> docs = null;
+    if(!preserveSchema.equals("Y")) {
+      docs = df.javaRDD().map(new Function<Row, SolrInputDocument>() {
+        public SolrInputDocument call(Row row) throws Exception {
+          StructType schema = row.schema();
+          SolrInputDocument doc = new SolrInputDocument();
+          for (StructField f : schema.fields()) {
+            String fname = f.name();
+            if (fname.equals("_version_"))
+              continue;
+
+            int fieldIndex = row.fieldIndex(fname);
+            Object val = row.isNullAt(fieldIndex) ? null : row.get(fieldIndex);
+            if (val != null)
+              doc.setField(fname, val);
+          }
+          return doc;
+        }
+      });
+    }
+    else{
+      docs = convertToSolrDocuments(df, new HashMap<String,Object>());
+    }
+
+    SolrSupport.indexDocs(solrRDD.zkHost, solrRDD.collection, 100, docs);
```

Comment:
sendToSolr(docs) was carried over from the previous implementation. The insert method already has the line "SolrSupport.indexDocs(solrRDD.zkHost, solrRDD.collection, 100, docs)" to send the docs to solr. I will remove the sendToSolr method.